docs/guides/force-organizations.mdx

```diff
@@ -148,134 +200,116 @@ export function SyncActiveOrganization() {
-        organization in order to proceed. If you are not part of an
-        organization, you can accept an invitation or create your own
-        organization
-      </p>
-      <OrganizationList hidePersonal={true} />
-    </section>
-  )
-}
-```
+<Tabs items={["Middleware", "Organization Selection page"]}>
+  <Tab>
+    ```tsx filename="middleware.ts"
+    import { authMiddleware, redirectToSignIn } from '@clerk/nextjs';
 
-    </Tab>
+    export default authMiddleware({
+      afterAuth(auth, req, evt) {
+        ...
+        // Redirect logged in users to organization selection page if they are not active in an organization
+        if(auth.userId && !auth.orgId && req.nextUrl.pathname !== "/org-selection"){
+          const searchParams = new URLSearchParams({
+            redirectUrl: req.url
+          })
 
-    <Tab>
+          const orgSelection = new URL(`/org-selection?${searchParams.toString()}`, req.url);
 
-    See the complete example of the middleware script ([here](/docs/references/nextjs/auth-middleware#use-after-auth-for-fine-grained-control))
-```tsx filename="middleware.ts"
-import { authMiddleware, redirectToSignIn } from '@clerk/nextjs';
- 
-export default authMiddleware({
-  afterAuth(auth, req, evt) {
-    ...
-    // Redirect logged in users to organization selection page if they are not active in an organization
-    if(auth.userId && !auth.orgId && req.nextUrl.pathname !== "/org-selection"){
-      const searchParams = new URLSearchParams({
-        redirectUrl: req.url
-      })
-      const orgSelection = new URL(`/org-selection?${searchParams.toString()}`, req.url);
-      return NextResponse.redirect(orgSelection)
+          return NextResponse.redirect(orgSelection)
+        }
+        ...
+      }
+    });
+    ```
+  </Tab>
+
+  <Tab>
+    ```tsx filename="pages/org-selection.tsx"
+    import { OrganizationList } from "@clerk/nextjs"
+    import { useRouter } from "next/router";
+
+    export default function OrganizationSelection() {
+      const { query } = useRouter();
+
+      return (
+        <section>
+          <h1>Welcome to the Organization Selection page.</h1>
+          <p>
+            This part of the application requires the user to select an
+            organization in order to proceed. If you are not part of an
+            organization, you can accept an invitation or create your own
+            organization.
+          </p>
+          <OrganizationList
+            hidePersonal={true}
+            afterCreateOrganizationUrl={query.redirectUrl ?? "/"}
+            afterSelectOrganizationUrl={query.redirectUrl ?? "/"}
+          />
+        </section>
+      )
     }
-    ...
-  }
-});
-```
+    ```
+  </Tab>
+</Tabs>
 
-```tsx filename="pages/org-selection.tsx"
-import { OrganizationList } from "@clerk/nextjs"
-import { useRouter } from "next/router";
+#### Limit parts of an application to only users with active organizations
```

Comment:
There is really limited different between 'whole application' and 'part of an application'

Whole application:
- check in app/layout.tsx
- or check in middleware
```
if(auth.userId && !auth.orgId && req.nextUrl.pathname !== "/org-selection"){
      const searchParams = new URLSearchParams({
        redirectUrl: req.url
      })
 
      const orgSelection = new URL(`/org-selection?${searchParams.toString()}`, req.url);
 
      return NextResponse.redirect(orgSelection)
```


Part of application
- check in app/dashboard/layout.tsx
- or check in middleware
```
if(auth.userId && !auth.orgId && <check if path is /dashboard> && req.nextUrl.pathname !== "/dashboard/org-selection"){
      const searchParams = new URLSearchParams({
        redirectUrl: req.url
      })
 
      const orgSelection = new URL(`/dashboard/org-selection?${searchParams.toString()}`, req.url);
 
      return NextResponse.redirect(orgSelection)
```

The guide could just cover these two options in one 'protect your application' general section.

Reply:
I've updated the organization to better reflect this information. Let me know what you think!